Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs

```diff
@@ -37,16 +39,21 @@ namespace QuantConnect.Brokerages.Bitfinex
     /// </summary>
     public partial class BitfinexBrokerage
     {
-        private const string ApiVersion = "v1";
+        private const string ApiVersion = "v2";
+        private const string RestApiUrl = "https://api.bitfinex.com";
+        private const string WebSocketUrl = "wss://api.bitfinex.com/ws/2";
+
         private readonly IAlgorithm _algorithm;
-        private readonly ConcurrentQueue<WebSocketMessage> _messageBuffer = new ConcurrentQueue<WebSocketMessage>();
-        private volatile bool _streamLocked;
-        private readonly RateGate _restRateLimiter = new RateGate(8, TimeSpan.FromMinutes(1));
+        private readonly RateGate _restRateLimiter = new RateGate(10, TimeSpan.FromMinutes(1));
         private readonly ConcurrentDictionary<int, decimal> _fills = new ConcurrentDictionary<int, decimal>();
         private readonly BitfinexSubscriptionManager _subscriptionManager;
         private readonly SymbolPropertiesDatabase _symbolPropertiesDatabase;
         private readonly IDataAggregator _aggregator;
 
+        // map ClientOrderId -> LEAN order
+        private readonly ConcurrentDictionary<long, Order> _orderMap = new ConcurrentDictionary<long, Order>();
```

Comment:
Seems a bit of duplication with `CachedOrderIDs`? We are not removing any order from this collection.
On `GetOpenOrders` I only see we update `CachedOrderIDs` if already present? Should we be adding them instead 🤔 same for `_orderMap`

Reply:
In `_orderMap` the key is not the Lean `OrderId` but the Bitfinex `ClientOrderId` field (`cid`) which is used to match async messages. 
In the docs they mention this field is meant to be unique in the (although not enforced):
https://docs.bitfinex.com/reference#ws-auth-input-order-new

I think using the `CachedOrderIDs` dictionary, we could have duplicate id issues with existing orders, if an algorithm is started twice on the same day, maybe should use a timestamp instead?